extensions/azurecore/src/azureResource/providers/sqlinstance/sqlInstanceTreeDataProvider.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionNodeType, TreeItem } from 'azdata';
+import { TreeItemCollapsibleState, ExtensionContext } from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+import { AzureResourceItemType } from '../../constants';
+import { ApiWrapper } from '../../../apiWrapper';
+import { generateGuid } from '../../utils';
+import { IAzureResourceService, AzureResourceDatabaseServer } from '../../interfaces';
+import { ResourceTreeDataProviderBase } from '../resourceTreeDataProviderBase';
+import { azureResource } from '../../azure-resource';
+
+export class SqlInstanceTreeDataProvider extends ResourceTreeDataProviderBase<AzureResourceDatabaseServer> {
+	private static readonly containerId = 'azure.resource.providers.sqlInstanceContainer';
+	private static readonly containerLabel = localize('azure.resource.providers.sqlInstanceContainerLabel', "SQL Instances");
+
+	public constructor(
+		databaseServerService: IAzureResourceService<AzureResourceDatabaseServer>,
+		apiWrapper: ApiWrapper,
+		private _extensionContext: ExtensionContext
+	) {
+		super(databaseServerService, apiWrapper);
+	}
+
+
+	protected getTreeItemForResource(databaseServer: AzureResourceDatabaseServer): TreeItem {
+		return {
+			id: `sqlInstance_${databaseServer.name}`,
+			label: databaseServer.name,
+			iconPath: {
+				dark: this._extensionContext.asAbsolutePath('resources/dark/sql_instance_inverse.svg'),
+				light: this._extensionContext.asAbsolutePath('resources/light/sql_instance.svg')
+			},
+			collapsibleState: TreeItemCollapsibleState.Collapsed,
+			contextValue: AzureResourceItemType.databaseServer,
```

Comment:
Ideally, we should keep `AzureResourceItemType.databaseServer` and `AzureResourceItemType.sqlInstance` separate. I'm assuming we have this contextValue so we can support the context menu actions. Would it make sense to add `AzureResourceItemType.sqlInstance.sqlInstance` to the `when` clause so that `sqlInstance` isn't confused with a normal `Server`?

Reply:
I guess so. My thinking here is this really is a server, but I can add a separate piece of logic and add to the union of supported node types if needed. Right now we just use Connect, and almost all support this anyhow so it was less pressing.